crates/lib/src/bootc_composefs/boot.rs

```diff
@@ -603,7 +627,12 @@ fn write_pe_to_esp(
 
     let pe_name = match pe_type {
         PEType::Uki => format!("{}{}", uki_id, EFI_EXT),
-        PEType::UkiAddon => format!("{}{}", uki_id, EFI_ADDON_FILE_EXT),
+        PEType::UkiAddon => file_path
+            .components()
+            .last()
+            .unwrap()
+            .to_string_lossy()
```

Comment:
Please no `to_string_lossy` in general except for display to the user. I think in this case we should be using `Utf8Path` here and propagate that constraint upwards.